File: 2657. Find the Prefix Common Array of Two Arrays.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) 
    {
        int n = A.size();
        vector<int> C(n), count(51);
        for(int i = 0; i < n; i++)
        {
            count[A[i]] += 1, count[B[i]] += 1;
            if(i == 0) C[i] = A[i] == B[i];
            else 
            {
                C[i] = C[i - 1];
                if(A[i] == B[i]) C[i] += 1;
                else C[i] += (count[A[i]] == 2) + (count[B[i]] == 2);
            }
        }
        return C;    
    }
};

File: 2657. Find the Prefix Common Array of Two Arrays.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        ans = []
        n = len(A)
        unique_vals = set()
        for i in range(n):
            unique_vals.add(A[i])
            unique_vals.add(B[i])

            count = 0 
            for target in unique_vals:
                count += target in A[:i+1] and target in B[:i+1]
            ans.append(count)    

        return ans